Team-C/test_wizard/lib/screens/login_screen.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+import 'package:test_wizard/services/auth_service.dart';
+
+class LoginScreen extends StatefulWidget {
+  const LoginScreen ({super.key});
+  @override
+  _LoginScreenState createState() => _LoginScreenState();
+}
+
+class _LoginScreenState extends State<LoginScreen> {
+  final AuthService _authService = AuthService();
+  final TextEditingController _usernameController = TextEditingController();
+  final TextEditingController _passwordController = TextEditingController();
+  bool _isLoading = false;
+
+  void _login() async {
+    setState(() {
+      _isLoading = true;
+    });
+
+    try {
+      final token = await _authService.authenticate(
+        _usernameController.text,
+        _passwordController.text,
+      );
+
+      if (token != null) {
+        Navigator.pushReplacementNamed(context, '/home');
+      }
+    } catch (e) {
+      ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(e.toString())));
+    } finally {
+      setState(() {
+        _isLoading = false;
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Login')),
+      body: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          children: [
+            TextField(
+              controller: _usernameController,
+              decoration: const InputDecoration(labelText: 'Username'),
+            ),
+            TextField(
+              controller: _passwordController,
+              decoration: const InputDecoration(labelText: 'Password'),
+              obscureText: true,
+            ),
+            const SizedBox(height: 20),
+            _isLoading
+                ? const CircularProgressIndicator()
+                : ElevatedButton(
+                    onPressed: _login,
+                    child: const Text('Login'),
+                  ),
```

Comment:
This looks ok if we are going to actually log in with a username and password, but @WhitneyMeulink keeps mentioning OAuth. @J-Gilberg @eshin2 @hellohouseplant @GhauriM what is the verdict on this? Can we just have a button that says "login with moodle" and moodle does the rest, or do we have to set it up like this, with username and password, and then send to the endpoint and receive a token back? Please advise, knowing this determines everything about login basically.

Reply:
What Lana has right now we should keep. This will work with our Moodle local instances and the downstream implementation. All that changes in moving to 0Auth 2 is where the token comes from. 

Created a ticket for investigating/implementing the configuration of 0Auth. This is something we could do last btw.

Whitney is not wrong 0Auth is what we called out in the TDD.

https://docs.moodle.org/404/en/OAuth_2_services